tools/state-viewer/src/latest_witnesses.rs

```diff
@@ -20,13 +44,17 @@ pub struct LatestWitnessesCmd {
     #[arg(long)]
     epoch_id: Option<EpochId>,
 
-    /// Pretty-print using the "{:#?}" formatting.
-    #[arg(long)]
-    pretty: bool,
+    #[clap(subcommand)]
+    /// Save mode
+    mode: LatestWitnessesMode,
+}
 
-    /// Print the raw &[u8], can be pasted into rust code
-    #[arg(long)]
-    binary: bool,
+#[derive(clap::Subcommand)]
+enum LatestWitnessesMode {
+    /// Pretty-print on screen using the "{:#?}" formatting.
+    Pretty,
+    /// Saves the raw &[u8] of each witness to the given directory.
+    Binary { output_dir: PathBuf },
```

Comment:
This is totally sufficient but if you fancy you can make it a bit quicker by adding some sort of selector to only pick the witnesses at desired height and shard. Totally optional as this PR is large enough already :)

Reply:
`DumpWitnessesCmd` (Latest) already supports filtering by height, shard and epoch. Or you mean giving height and shard to `Validate` command? But it supports only one witness anyway.

Reply:
Ah, sorry, I missed that. It's fine as is, thanks!